src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -0,0 +1,283 @@
+    estimated as a quantile of weekly variance in metrics. Sample
+    size calculation is based on the asymptotic relative efficiency (ARE) of
+    the U test to the T test (see Stapleton 2008, pg 266, or
+    https://www.psychologie.hhu.de/fileadmin/redaktion/Fakultaeten/
+    Mathematisch-Naturwissenschaftliche_Fakultaet/Psychologie/AAP/gpower/GPowerManual.pdf)
+
+    Args:
+        res: A TimeSeriesResult, generated by
+            mozanalysis.sizing.HistoricalTarget.get_time_series_data.
+        bq_context: A mozanalysis.bq.BigQueryContext object that handles downloading
+            time series data from BigQuery.
+        metrics_list (list of mozanalysis.metrics.Metric): List of metrics
+            used to construct the results df from HistoricalTarget. The names
+            of these metrics are used to return results for sample size
+            calculation for each.
+        quantile (float, default .90): Quantile used to calculate the effect size
+            as the quantile of week-to-week metric changes and the variance of
+            the mean.
+        alpha (float, default .05): Significance level for the experiment.
+        power (float, default .90): Probability of detecting an effect,
+            when a significant effect exists.
+        parent_distribution (str, default "normal"): Distribution of the parent data;
+            must be normal, uniform, logistic, or laplace.
+
+    Returns a sorted dictionary:
+        Keys in the dictionary are the metrics column names from the DataFrame; values
+        are the required sample size to achieve the desired power for that metric.
+        The dictionary is sorted by required sample size, in descending order.
+    """
+
+    def _mann_whitney_solve_sample_size_approximation(
+        effect_size,
+        std,
+        alpha=.05,
+        power=.8,
+        parent_distribution="normal"
+    ):
+
+        """
+        Wilcoxen-Mann-Whitney rank sum test sample size calculation,
+        based on asymptotic efficiency relative to the t-test.
+        """
+        rel_effect_size = effect_size/std
+        are = {
+            "uniform": 1.,
+            "normal": pi/3.,
+            "logistic": 9./(pi**2),
+            "laplace": 2./3.
+        }
+
+        if parent_distribution not in are.keys():
+            raise ValueError(f"Parent distribution must be in {are.keys()}")
+
+        t_sample_size = tt_ind_solve_power(
+            effect_size=rel_effect_size,
+            power=power,
+            alpha=alpha
+        )
+
+        return t_sample_size*are[parent_distribution]
+
+    res_mean = res.get_aggregated_data(
+        bq_context=bq_context,
+        metric_list=metric_list,
+        aggregate_function="AVG"
+    )
+
+    res_mean.sort_values(by="analysis_window_start", ascending=True, inplace=True)
+    effect_size = {
+        m.name: res_mean[m.name].diff().quantile(q=quantile) for m in metric_list
```

Comment:
```suggestion
        m.name: res_mean[m.name].diff().abs().quantile(q=quantile) for m in metric_list
```
Could we take the absolute value of differences?

Reply:
Yup! I made that change in the most recent commit